rest/config.go

```diff
@@ -355,6 +355,10 @@ func (dbConfig *DbConfig) AutoImportEnabled() (bool, error) {
 }
 
 func (dbConfig *DbConfig) validate() []error {
+	return dbConfig.validateVersion(base.IsEnterpriseEdition())
```

Comment:
There are more distinctions between CE and EE than just this flag (like some private librares we pull in, functional differences in the code, etc.), so all of the current tests can be run in both CE and EE using the Go build tag: `cb_sg_enterprise`.

With this in mind, I'm not sure this change is required for testing both CE and EE dbConfigs. You can call `base.IsEnterpriseEdition()` from the test itself to do different assertions. A good example of this can be found in: `TestConfigValidationCache`

Reply:
Thanks @bbrks for the feedback. The intention of this approach is to cover DBConfig validation behavior for both CE and EE scenarios during the test irrespective of the actual value returned from the node through base.IsEnterpriseEdition(). When introducing validateVersion method on DbConfig instance, it would be easy to simulate the CE and EE scenarios by passing a boolean value (true for Enterprise Edition and false for Community Edition) during the test runs. Also both validate and validateVersion methods can be covered by this approach.
func (dbConfig *DbConfig) validate() []error {
	return dbConfig.validateVersion(base.IsEnterpriseEdition())
}
func (dbConfig *DbConfig) validateVersion(isEnterpriseEdition bool) []error {
...
}
When looking at TestConfigValidationCache; the test is relying on the actual sync gateway version returned from the node to decide which all assertions need to be executed. But in the above approach both cases will be covered during the run since the values are mocked up.
if base.IsEnterpriseEdition() {
		require.NotNil(t, config.Databases["db"].CacheConfig.RevCacheConfig.Size)
		assert.Equal(t, 0, int(*config.Databases["db"].CacheConfig.RevCacheConfig.Size))
	} else {
		// CE disallowed - should be nil
		assert.Nil(t, config.Databases["db"].CacheConfig.RevCacheConfig.Size)
}
Would you be able to check this once again and let me know if you agree with my reasoning; If you disagree and still think that the above approach is a bad choice we can totally remove the changes. Please let me know. (edited)

Reply:
Your reasoning is valid for this scenario, if we wanted to see coverage for both CE/EE in only a single edition's run. I'm happy to leave it as is given the function's narrow responsibility.

We just need to be careful if we use this approach in other areas of code, as the functional differences between CE/EE aren't as easily changed, as well as potential side effects from the CE/EE differences.